android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonsPermissionsAdapterTest.kt

```diff
@@ -9,12 +9,12 @@ import android.widget.TextView
 import androidx.core.content.ContextCompat
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import mozilla.components.feature.addons.R
 import mozilla.components.feature.addons.ui.AddonPermissionsAdapter.PermissionViewHolder
 import mozilla.components.feature.addons.ui.AddonPermissionsAdapter.Style
 import mozilla.components.support.test.mock
 import mozilla.components.support.test.robolectric.testContext
 import mozilla.components.support.test.whenever
+import mozilla.components.ui.colors.R
 import org.junit.Test
```

Comment:
Was this part of the automatic conversion tool? There's a whole bunch of similar instances throughout the PR. Any idea why these got changed?

(Correct me if I'm wrong, but I thought both of these represent non-transitive resource path references, so I wouldn't expect there to be a performance difference. If that's the case, I tend to get a little nervous about automated changes whose purpose we don't have a clear justification for)

So, essentially:

1. Does this risk changing any runtime behavior? (e.g. referencing the wrong resource)
2. Does this actually add any value, or was it just automatically updated? (And, if so, do we know why the tool picked these changes?)

Reply:
The parts where the import name changes are manual changes that the automatic tool actually missed; these changes point the `R` reference to the proper class whereas before they were pointing to a different class that just so happened to pull in the correct `R` resource(s) upon dependency resolution (which no longer occurs for non-transitive `R` classes).

> Correct me if I'm wrong, but I thought both of these represent non-transitive resource path references

The removed import is actually a transitive reference as the `addons` module's `R` doesn't directly contain the resource (in this case the color `photonBlue40`), and instead depends on the `ui-colors` module that contains the `R` class that contains the color resource.

Before: `AddonsPermissionsAdapterTest` -> `addons R` -> `ui-colors R` -> `photonBlue40`
After: `AddonsPermissionsAdapterTest` -> `ui-colors R` -> `photonBlue40`

> Does this risk changing any runtime behavior? (e.g. referencing the wrong resource)

No, the changes were required to compile successfully with `android.nonTransitiveRClass=true`. I also checked that there were no other possible resources that the statements could have been referencing instead.

Reply:
Ahh, that all makes sense, I was indeed misinterpreting the context of non-transitive R classes. Appreciate the thorough rundown.